e107_plugins/download/download_shortcodes.php

```diff
@@ -691,13 +695,18 @@ function sc_download_view_link($parm)
 
 		$click = "";
 		
-		$img = "<img src='".IMAGE_DOWNLOAD."' alt='".LAN_DOWNLOAD."' title='".LAN_DOWNLOAD."' />";
       
-		if(deftrue('BOOTSTRAP'))
-		{
-			$img = e107::getParser()->toGlyph('download',$parm); // '<i class="icon-download"></i>'; 
-		}	
-		
+		switch (deftrue('BOOTSTRAP')) {
+				case 4:
+						$img = e107::getParser()->toGlyph('fa-download',false);
+						break;
+				case 3:
+						$img = e107::getParser()->toGlyph('icon-download.glyph',false);
+						break;
+				default:
+						$img = "<img src='".IMAGE_DOWNLOAD."' alt='".LAN_DOWNLOAD."' title='".LAN_DOWNLOAD."' />";
+						break;
```

Comment:
Perhaps using `toIcon()` would be more efficient. When it detects the .glyph extension it automatically uses` toGlyph()` otherwise, it returns an `<img>`. ie. you could send it `fa-download.glyph`, `icon-download.glyph` or `{e_PATHXX}whatever.png`

Reply:
It was written this way before, I just added switch. I need an example where this is used correct way and I can repeat it, but I am not able to write it myself.

Reply:
You should be able to find toIcon() used throughout the core in various ways. Please let me know if you can't, and I'll do a search for you.

Reply:
yes, but I wrote you somewhere that that code looks reductant because it's already in shortcodes.